rocketpy/tools.py

```diff
@@ -515,44 +515,49 @@ def generate_monte_carlo_ellipses_coordinates(
 
     Returns
     -------
-    list
+    list[list[tuple[float, float]]]
         List of lists of tuples containing the latitude and longitude of each
         point in each ellipse.
     """
-    outputs = [None] * len(ellipses)
-
-    for index, ell in enumerate(ellipses):
-        # Get ellipse path points
-        center = ell.get_center()
-        width = ell.get_width()
-        height = ell.get_height()
-        angle = np.deg2rad(ell.get_angle())
-        points = lat_lon_points = [None] * resolution
-
-        # Generate ellipse path points (in a Cartesian coordinate system)
-        for i in range(resolution):
-            x = width / 2 * math.cos(2 * np.pi * i / resolution)
-            y = height / 2 * math.sin(2 * np.pi * i / resolution)
-            x_rot = center[0] + x * math.cos(angle) - y * math.sin(angle)
-            y_rot = center[1] + x * math.sin(angle) + y * math.cos(angle)
-            points[i] = (x_rot, y_rot)
-        points = np.array(points)
-
-        # Convert path points to lat/lon
-        for point in points:
-            x, y = point
-            # Convert to distance and bearing
-            d = math.sqrt((x**2 + y**2))
-            bearing = math.atan2(
-                x, y
-            )  # math.atan2 returns the angle in the range [-pi, pi]
-
-            lat_lon_points[i] = inverted_haversine(
-                origin_lat, origin_lon, d, bearing, earth_radius=6.3781e6
-            )
-
-        outputs[index] = lat_lon_points
-    return outputs
+    return [
+        __convert_to_lat_lon(
+            __generate_ellipse_points(ell, resolution), origin_lat, origin_lon
+        )
+        for ell in ellipses
+    ]
+
+
+def __convert_to_lat_lon(points: list, origin_lat: float, origin_lon: float):
+    return [
+        inverted_haversine(
+            origin_lat,
+            origin_lon,
+            math.sqrt(x**2 + y**2),
+            math.degrees(math.atan2(x, y)),
+            earth_radius=6.3781e6,
+        )
+        for x, y in points
+    ]
+
+
+def __generate_ellipse_points(ellipse, resolution: int):
```

Comment:
Small type hint. I do not oppose adding them in a module whole module, but since this is the only parameter that has it, I believe it is a "typo".